src/dbnode/encoding/iterator_pool.go

```diff
@@ -65,5 +68,6 @@ func (p *multiReaderIteratorPool) Get() MultiReaderIterator {
 }
 
 func (p *multiReaderIteratorPool) Put(iter MultiReaderIterator) {
+	iter.SetSchema(nil)
```

Comment:
This should just be handled by `Reset(...)` then you don't need to add extra logic to the pool gets/puts.

Making pools know more about their data structures is a pattern we've tried to avoid, encapsulating that logic in `Reset(...)` for each separate data structure helps make these pools simple.

Reply:
ditto